core-common/src/main/java/org/mapstruct/MapMapping.java

```diff
@@ -55,10 +55,21 @@
     String valueDateFormat() default "";
 
     /**
-     * A key value qualifier can be specified to aid the selection process of a suitable mapper. This is useful in
-     * case multiple mappers (hand written of internal) qualify and result in an 'Ambiguous mapping methods found'
-     * error.
+     * A qualifier can be specified to aid the selection process of suitable {@link BeforeMapping} /
+     * {@link AfterMapping} methods.
+     * <p>
+     * Use {@link #keyQualifiedBy()} / {@link #valueQualifiedBy()} to specify qualifiers to be used for selecting key /
+     * value mapping methods.
      *
+     * @return the qualifiers
+     */
+    Class<? extends Annotation>[] qualifiedBy() default { };
```

Comment:
Hum, I think that's quite confusing, given that there are `keyQualifiedBy()` and `valueQualifiedBy()` already. Is the application of qualifiers actually needed for the lifecycle methods? Or is this something which we could leave out, at least for now?